zone/client.cpp

```diff
@@ -10276,13 +10276,14 @@ void Client::RemoveItem(uint32 item_id, uint32 quantity)
 
 			item = GetInv().GetItem(slot_id);
 			if (item && item->GetID() == item_id) {
-				int stack_size = item->IsStackable() ? item->GetCharges() : 1;
-				if ((removed_count + stack_size) <= quantity) {
-					removed_count += stack_size;
+				int stack_size = item->IsStackable() ? item->GetCharges() : 0;
+				int real_stack_size = stack_size ? stack_size : 1;
```

Comment:
Can do `= std::max(stack_size, 1)` to make this a little clearer
Maybe also just rename `stack_size` to `charges` and then this `real_stack_size` to the original `stack_size` for clarity